src/main/java/uk/gov/hmcts/darts/audio/mapper/GetTransformedMediaDetailsMapper.java

```diff
@@ -0,0 +1,67 @@
+import uk.gov.hmcts.darts.common.entity.TransformedMediaEntity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class GetTransformedMediaDetailsMapper {
+    public List<SearchTransformedMediaResponse> mapSearchResults(List<TransformedMediaEntity> entityList) {
+        List<SearchTransformedMediaResponse> mappedDetails = new ArrayList<>();
+        for (TransformedMediaEntity entity : entityList) {
+            SearchTransformedMediaResponse transformedMediaDetails = new SearchTransformedMediaResponse();
+            transformedMediaDetails.setId(entity.getId());
+            transformedMediaDetails.setFileName(entity.getOutputFilename());
+
+            if (entity.getOutputFormat() != null) {
+                transformedMediaDetails.setFileFormat(entity.getOutputFormat().name());
+            }
+
+            transformedMediaDetails.setLastAccessedAt(entity.getLastAccessed());
+            transformedMediaDetails.setFileSizeBytes(entity.getOutputFilesize());
+
+            if (entity.getMediaRequest() != null) {
+                SearchTransformedMediaResponseMediaRequest mediaRequest = new SearchTransformedMediaResponseMediaRequest();
+                mediaRequest.setId(entity.getMediaRequest().getId());
+                mediaRequest.setRequestedAt(entity.getMediaRequest().getCreatedDateTime());
+                mediaRequest.setOwnerUserId(entity.getMediaRequest().getCurrentOwner().getId());
+                mediaRequest.setRequestedByUserId(entity.getMediaRequest().getRequestor().getId());
+                transformedMediaDetails.setMediaRequest(mediaRequest);
+            }
+
+            if (entity.getMediaRequest() != null && entity.getMediaRequest().getHearing() != null
+                && entity.getMediaRequest().getHearing().getCourtCase() != null) {
+                SearchTransformedMediaResponseCase caseResponse = new SearchTransformedMediaResponseCase();
+                caseResponse.setId(entity.getMediaRequest().getHearing().getCourtCase().getId());
+                caseResponse.setCaseNumber(entity.getMediaRequest().getHearing().getCourtCase().getCaseNumber());
+                transformedMediaDetails.setCase(caseResponse);
+            }
+
+            if (entity.getMediaRequest() != null && entity.getMediaRequest().getHearing() != null
+                && entity.getMediaRequest().getHearing().getCourtroom() != null) {
+                SearchTransformedMediaResponseCourthouse courthouseResponse = new SearchTransformedMediaResponseCourthouse();
+                courthouseResponse.setId(entity.getMediaRequest().getHearing().getCourtroom().getCourthouse().getId());
+                courthouseResponse.setDisplayName(entity.getMediaRequest().getHearing().getCourtroom().getCourthouse().getDisplayName());
+                transformedMediaDetails.setCourthouse(courthouseResponse);
+            }
+
+            if (entity.getMediaRequest() != null && entity.getMediaRequest().getHearing() != null) {
```

Comment:
`entity.getMediaRequest() != null && entity.getMediaRequest().getHearing() != null` is in every one of the previous if-statements, so should we be trying to combine them?